### DISCUSSION TITLE
Repository Pattern

### DISCUSSION THREAD

**Discussion by alejojperez** [laravel/nova-issues#4167] [General]
I think that Laravel Nova has tremendous potential to be a really good tool to build enterprise applications. Having said that, when thinking about enterprise tools and scalability the Repository Pattern is one of the first things that come to mind. It would be a very good addition to include an easy way to use the repository pattern on Nova Resources instead of the eloquent model directly.

**Comment by ziming** [comment]
instead of repositories consider this

https://stitcher.io/blog/laravel-beyond-crud-04-models

it's way better in my opinion
